base/src/gpio/gpio.hpp

```diff
@@ -0,0 +1,129 @@
+#ifndef GPIO_HPP
+#define GPIO_HPP
+
+#include "../system/stm32h750xx.h"
+#include "gpio_defs.h"
+
+namespace gpio{
+
+#define NUM_GPIO_PORTS 11
+
+class GPIOPort{
+public:
+    GPIOPort(GPIO_TypeDef* mPort);
+
+    void setPinData(uint8_t pin, bool data);
+    bool getPinData(uint8_t pin);
+    void setPinMode(uint8_t pin, eGPIOPinMode mode);
+    void setPinSpeed(uint8_t pin, eGPIOPinSpeed speed);
+    void setPinType(uint8_t pin, eGPIOPinType speed);
+    void setPinPull(uint8_t pin, eGPIOPinPull pull);
+    void setPinAlt(uint8_t pin, uint8_t alt);
+
+    void setParallelData(uint16_t data);
+    uint16_t getParallelData();
+
+private:
+
+    GPIO_TypeDef* mPort; ///< GPIO Port i.e. GPIOA
+    uint8_t mPortNum;    ///< GPIO port num i.e A is 0
+};
+
+class Pin{
+public:
+
+    ///
+    /// Constructor.
+    ///
+    Pin(GPIOPort* port, uint8_t pin):
+    mPort(port),
+    mPin(pin)
+    {;} // Do nothing in constructor.
+
+    void operator = (bool value){
+        mPort->setPinData(mPin, value);
+    }
+
+    bool operator () (){
+        return mPort->getPinData(mPin);
+    }
+
+private:
+    GPIOPort* mPort; //< Ponter to port
```

Comment:
Spelling mistake

Reply:
Modified.

Reply:
Removed, the variable was for debug purposes.